### DISCUSSION TITLE
How should I snapshot and rollback both root and home together?

### DISCUSSION THREAD

**Discussion by pycaw** [zbm-dev/zfsbootmenu#699] [Questions]
My goal is that I want to ensure snapshots that contain both root and home filesystems and that they are in sync.

The [docs ](https://docs.zfsbootmenu.org/en/v2.3.x/general/bootenvs-and-you.html) say the followings:

> When the filesystems are on different snapshot schedules (or there is some delay between snapshotting one after the other), deciding which snapshots represent consistent state may not be a trivial task.

> ...remedied with a recursive snapshot scheme that provides uniform nomenclature for consistent snapshots across multiple filesystems

>  If you wish to maintain more complicated setups, you can always manually manage snapshot rollbacks or clone operations from the recovery shell that ZFSBootMenu provides.

I followed the Fedora guide, so I have zroot/ROOT/$ID at `/` and zroot/home at `/home`.

So based on this, could I achieve the workflow I imagine by,
```
zfs snapshot -r  zroot/ROOT/$ID@snapshot-$(date +%Y.%m.%d_%H-%M-%S)
```
or
```
SNAPSHOT_NAME="snapshot-$(date +%Y.%m.%d_%H-%M-%S)"
zfs snapshot zroot/ROOT/$ID@$SNAPSHOT_NAME
zfs snapshot zroot/home@$SNAPSHOT_NAME
```

and then,
```
SNAPSHOT_NAME=...
zfs rollback zroot/root/$ID@$SNAPSHOT_NAME
zfs rollback zroot/home@$SNAPSHOT_NAME
```
?

**Comment by ahesford** [comment]
You almost never want to rollback your home directory when you rollback a boot environment. That's the primary reason for making your home directory a separate filesystem; the two are notionally independent.

In any case, ZFSBootMenu will accommodate any snapshot policy you like. We take no position on particular strategies and recommend you consult general-purpose ZFS forums for advice and discussion about best practices for snapshots.

**Reply by pycaw** [reply]
They are independent, to a degree... Of course it makes sense to back them up together, essentially the OS. I could have different OS'es then offered by ZFSBootMenu.
